src/app/screens/Dashboard/screens/Detail/index.tsx

```diff
@@ -11,8 +13,20 @@ function Detail() {
   const { category, component } = useParams();
   const { data, loading } = useQuery(getComponentFiles(category as string, component as string));
   const componentCode = getComponentCode(data?.repository);
+  
+  const downloadZip = useCallback(() => {
+    const { html, scss, css } = componentCode;
```

Comment:
I think you should loop through the keys of the `componentCode`, that way you can get all files and that should support React code as well (maybe a few tweaks will be needed 😝)

Reply:
How do i manage the extension in the file?

Reply:
Touchè. Maybe you can use the key name? Ok, somehow we need to add the name to the files so.. that's something to be added later on. For now it's ok I think 😄